lib/matplotlib/backends/qt_editor/formsubplottool.py

```diff
@@ -1,230 +1,56 @@
-        self.sliderwspace.setMouseTracking(False)
-        self.sliderwspace.setProperty("value", 0)
-        self.sliderwspace.setOrientation(QtCore.Qt.Horizontal)
-        self.sliderwspace.setInvertedAppearance(False)
-        self.sliderwspace.setInvertedControls(False)
-        self.sliderwspace.setTickPosition(QtWidgets.QSlider.TicksAbove)
-        self.sliderwspace.setTickInterval(100)
-
-        self.wspacevalue = QtWidgets.QLabel('0', self)
-        self.wspacevalue.setMinimumSize(QtCore.QSize(30, 0))
-        self.wspacevalue.setAlignment(
-            QtCore.Qt.AlignRight |
-            QtCore.Qt.AlignTrailing |
-            QtCore.Qt.AlignVCenter)
-
-        self.verticalLayout.addLayout(hboxwspace)
-        hboxwspace.addWidget(self.labelwspace)
-        hboxwspace.addWidget(self.sliderwspace)
-        hboxwspace.addWidget(self.wspacevalue)
-
-        # button bar
-        hbox2 = QtWidgets.QHBoxLayout()
-        gbox.addLayout(hbox2, 8, 0, 1, 1)
-        self.tightlayout = QtWidgets.QPushButton('Tight Layout', self)
-        spacer = QtWidgets.QSpacerItem(
-            5, 20, QtWidgets.QSizePolicy.Expanding,
-            QtWidgets.QSizePolicy.Minimum)
-        self.resetbutton = QtWidgets.QPushButton('Reset', self)
-        self.donebutton = QtWidgets.QPushButton('Close', self)
-        self.donebutton.setFocus()
-        hbox2.addWidget(self.tightlayout)
-        hbox2.addItem(spacer)
-        hbox2.addWidget(self.resetbutton)
-        hbox2.addWidget(self.donebutton)
-
-        self.donebutton.clicked.connect(self.accept)
+        self.setObjectName("SubplotTool")
+        self._widgets = {}
+
+        layout = QtWidgets.QHBoxLayout()
+        self.setLayout(layout)
+
+        left = QtWidgets.QVBoxLayout()
+        layout.addLayout(left)
+        right = QtWidgets.QVBoxLayout()
+        layout.addLayout(right)
+
+        box = QtWidgets.QGroupBox("Borders")
+        left.addWidget(box)
+        inner = QtWidgets.QFormLayout(box)
+        for side in ["top", "bottom", "left", "right"]:
+            self._widgets[side] = widget = QtWidgets.QDoubleSpinBox()
+            widget.setMinimum(0)
+            widget.setMaximum(1)
+            widget.setDecimals(3)
+            widget.setSingleStep(.005)
+            widget.setKeyboardTracking(False)
+            inner.addRow(side, widget)
+        left.addStretch(1)
+
+        box = QtWidgets.QGroupBox("Spacings")
+        right.addWidget(box)
+        inner = QtWidgets.QFormLayout(box)
+        for side in ["hspace", "wspace"]:
+            self._widgets[side] = widget = QtWidgets.QDoubleSpinBox()
+            widget.setMinimum(0)
+            widget.setMaximum(1)
+            widget.setDecimals(3)
+            widget.setSingleStep(.005)
+            widget.setKeyboardTracking(False)
+            inner.addRow(side, widget)
+        right.addStretch(1)
+
+        self._widgets["Export values"] = widget = \
+            QtWidgets.QPushButton("Export values")
```

Comment:
We try to avoid backslash continuations, and avoiding this one is easy:
```python
widget = QtWidgets.QPushButton("Export values")
self._widgets["Export values"] = widget
```